commit fcca22a7c7b30ecfe337ba57265019e63e22acf3
Author: leskal <alexakis1997@gmail.com>
Date:   Sun Mar 19 17:11:28 2017 +0200

    Fix tether with ap0 interface
    
    Change-Id: Ie37948f00c3254c67b4fcbe37b9b6a5212804573

diff --git a/server/SoftapController.cpp b/server/SoftapController.cpp
index 1c59ec3..99830f7 100755
--- a/server/SoftapController.cpp
+++ b/server/SoftapController.cpp
@@ -54,7 +54,11 @@ using android::base::WriteStringToFile;
 #endif
 
 #ifdef LIBWPA_CLIENT_EXISTS
+#ifndef MTK_HARDWARE
 std:: string hostapd_unix_file(StringPrintf("/data/misc/wifi/hostapd/wlan0"));
+#else
+std:: string hostapd_unix_file(StringPrintf("/data/misc/wifi/hostapd/ap0"));
+#endif
 static const char HOSTAPD_DHCP_DIR[]    = "/data/misc/dhcp";
 #endif
 static const char HOSTAPD_CONF_FILE[]    = "/data/misc/wifi/hostapd.conf";
@@ -289,7 +293,12 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
             "hw_mode=%c\n"
             "ignore_broadcast_ssid=%d\n"
             "wowlan_triggers=any\n",
-            argv[2], argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
+#ifndef MTK_HARDWARE
+            argv[2]
+#else
+            "ap0"
+#endif
+                , argv[3], channel, (channel <= 14) ? 'g' : 'a', hidden));
 
     std::string fbuf;
     if (argc > 7) {
